swatch-billable-usage/ct/java/tests/TallySummaryConsumerComponentTest.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package tests;
+
+import static api.BillableUsageTestHelper.createTallySummaryWithDefaults;
+import static com.redhat.swatch.component.tests.utils.Topics.BILLABLE_USAGE;
+import static com.redhat.swatch.component.tests.utils.Topics.TALLY;
+
+import api.MessageValidators;
+import org.candlepin.subscriptions.billable.usage.TallySummary;
+import org.junit.jupiter.api.Test;
+
+public class TallySummaryConsumerComponentTest extends BaseBillableUsageComponentTest {
+
+  /** Verify tally summary is processed and billable usage is produced */
+  @Test
+  public void testBasicTallyToBillableUsageFlow() {
+    // Setup
+    String productId = "rhel-for-x86-els-payg";
+    String metricId = "vCPUs";
+    double value = 4.0;
+
+    // Send TallySummary to tally topic
+    TallySummary tallySummary = createTallySummaryWithDefaults(orgId, productId, metricId, value);
+    kafkaBridge.produceKafkaMessage(TALLY, tallySummary);
+
+    // Verify billable usage is produced
+    kafkaBridge.waitForKafkaMessage(
+        BILLABLE_USAGE, MessageValidators.billableUsageMatches(orgId, productId), 1);
+  }
+
+  /** Verify billable usage correctly accounts for contract coverage */
+  @Test
+  public void testBillableUsageWithContractCoverage() {
+    // Setup
+    String productId = "rhel-for-x86-els-payg";
+    String metricId = "vCPUs";
+    double contractCoverage = 2.0;
+    double totalUsage = 4.0;
+
+    // Setup contract to cover 2.0 vCPUs
+    contractsWiremock.setupContractCoverage(orgId, productId, metricId, contractCoverage);
+
+    // Send TallySummary with 4.0 vCPUs total usage
+    TallySummary tallySummary =
+        createTallySummaryWithDefaults(orgId, productId, metricId, totalUsage);
+    kafkaBridge.produceKafkaMessage(TALLY, tallySummary);
+
+    // Verify billable usage is produced (should account for the 2.0 vCPU coverage)
+    kafkaBridge.waitForKafkaMessage(
+        BILLABLE_USAGE, MessageValidators.billableUsageMatches(orgId, productId), 1);
```

Comment:
same here, I would assert that the billable usage takes into account the contract coverage in the matched billable usage.